libs/use-viewer/src/index.ts

```diff
@@ -89,127 +36,204 @@ const useViewer = ({ streamName, streamAccountId, subscriberToken, handleError }
     }
   };
 
-  // useEffect(() => {
-  //   switch (viewerState) {
-  //     case 'liveOff':
-  //       viewer.current?.webRTCPeer?.stopStats();
-  //       viewer.current?.webRTCPeer?.removeListener('stats', statisticsEventHandler);
-  //       break;
-  //     case 'liveOn':
-  //       {
-  //         viewer.current?.webRTCPeer?.initStats();
-  //         viewer.current?.webRTCPeer?.addListener('stats', statisticsEventHandler);
-  //       }
-  //       break;
-  //     default:
-  //       break;
-  //   }
-  // }, [viewerState]);
-
-  // const updateSourceQuality = (sourceId: SourceId, quality: StreamQuality) => {
-  //   if (!viewer.current) return;
-  //   const option = streamQualityOptions.find((option) => option.streamQuality === selectedQuality);
-  //   if (!option) return;
-  //   streamQuality.current = selectedQuality;
-  //   if (selectedQuality === 'Auto') {
-  //     viewer.current.select({});
-  //   } else {
-  //     viewer.current.select(option.simulcastLayer);
-  //   }
-  // };
-
-  const broadcastEventHandler = useCallback(async (event: BroadcastEvent) => {
-    console.log('got event');
-    if (!viewer.current) return;
-    console.log('broadcast event', event);
+  const connect = async () => {
+    const { current: viewer } = viewerRef;
+
+    if (!viewer) {
+      return;
+    }
+
+    let numReconnectionAttempts = 0;
+
+    try {
+      await viewer.connect({ events: ['active', 'inactive', 'layers', 'viewercount'] });
+    } catch (connectError) {
+      handleInternalError(connectError);
+
+      // eslint-disable-next-line no-constant-condition
+      while (true) {
```

Comment:
reconnect function will handle retry in an internal loop, so we should remove the `while` here